force_bdss/core/verifier.py

```diff
@@ -38,13 +36,26 @@ def _check_mco(workflow):
         errors.append(VerifierError(subject=mco,
                                     error="MCO has no defined parameters"))
 
-    for param in mco.parameters:
+    for idx, param in enumerate(mco.parameters):
         if len(param.name.strip()) == 0:
             errors.append(VerifierError(subject=param,
-                                        error="Parameter has empty name"))
+                                        error="Parameter {} "
+                                              "has empty name".format(idx)))
         if len(param.type.strip()) == 0:
             errors.append(VerifierError(subject=param,
-                                        error="Parameter has empty type"))
+                                        error="Parameter {} "
+                                              "has empty type".format(idx)))
+
+    for idx, kpi in enumerate(mco.kpis):
+        if len(kpi.name.strip()) == 0:
+            errors.append(VerifierError(subject=kpi,
+                                        error="KPI {} has empty name".format(
+                                            idx)))
+        if len(kpi.objective.strip()) == 0:
```

Comment:
This is an enumeration, it can never be empty unless we add the empty string as an entry in the enum itself.

Reply:
I think with KPIs the name is initially set to be empty, or if you remove the datasource providing the output designated as a KPI.
Similarly for the kpi objective I thought it made sense to have the empty string as a default option, as a guard against someone running a simulation with a viable but unhelpful objective. e.g. minimising when they needed to maximise. I once wasted ~8 days of compute time because I did something like this!

Reply:
ok, so what you need is Enum('', "MINIMISE"). The first argument is always the default. You don't need to treat it as a string.